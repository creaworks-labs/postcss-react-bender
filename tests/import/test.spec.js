var { processFile } = require("../utils");

describe('at-rule @import directives', () => {
  it("process complex definitions with @import directive", async () => {
    const fixture = require.resolve('./fixtures/test.scss')

    const result = await processFile(fixture);

    expect(result.bender).toMatchSnapshot();
    expect(result.warnings().length).toBe(0);
  });
})