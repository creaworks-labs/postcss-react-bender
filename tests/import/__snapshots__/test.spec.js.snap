// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`at-rule @import directives process complex definitions with @import directive 1`] = `
Object {
  ".test": Object {
    "alignItems": "center",
    "backgroundColor": "yellow",
    "fontFamily": "Georgia, serif",
    "fontSize": 12,
    "fontStyle": "italic",
    "fontVariant": Array [
      "small-caps",
    ],
    "fontWeight": "500",
    "height": 250,
    "justifyContent": "center",
    "lineHeight": 30,
    "marginTop": "ali",
    "width": 250,
  },
  "Screen": Object {
    ".square2": Object {
      "alignItems": "center",
      "backgroundColor": "yellow",
      "height": 250,
      "justifyContent": "center",
      "marginTop": "ali",
      "width": 250,
    },
    "View": Object {
      "View": Object {
        ".circle": Object {
          "paddingHorizontal": 20,
        },
        ".nestedCircle": Object {
          "overflow": "visible",
        },
        ".square": Object {
          "alignItems": "center",
          "backgroundColor": "yellow",
          "height": 250,
          "justifyContent": "center",
          "marginTop": "ali",
          "width": 250,
        },
      },
    },
    "container": Object {
      "alignItems": "center",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "marginTop": 50,
    },
    "input": Object {
      "backgroundColor": "red",
      "height": 50,
      "width": 100,
    },
    "title": Object {
      "color": "navy",
      "fontSize": 18,
    },
  },
  "circle": Object {
    "backgroundColor": "#fff",
    "borderBottomLeftRadius": 100,
    "borderBottomRightRadius": 100,
    "borderTopLeftRadius": 100,
    "borderTopRightRadius": 100,
    "height": 100,
    "width": 100,
  },
}
`;
